File: Aplicación Escritorio/ModeloTDO/UsuarioDTO.cs
using Modelo;
using System.Text.Json.Serialization;

namespace ModeloDTO
{
    public class UsuarioDTO
    {
        public long idUsuario { get; set; }
        public String nombre { get; set; }
        public String apellido { get; set; }
        public String email { get; set; }
        public String password { get; set; }

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public TIPOUSUARIO? rol { get; set; }
    }
}

File: Springboot_API/RetaCantabria/src/main/java/org/example/DTO/UsuarioDTO.java
package org.example.DTO;

import org.example.Entidades.TIPOUSUARIO;
import org.example.Entidades.Usuario;

public record UsuarioDTO(
        Long idUsuario,
        String nombre,
        String apellido,
        String email,
        String password,
        TIPOUSUARIO rol
) {
    public UsuarioDTO(Usuario usu){
        this(
                usu.getIdUsuario(),
                usu.getNombre(),
                usu.getApellido(),
                usu.getEmail(),
                usu.getPassword(),
                usu.getRol()
        );
    }
}
